var/spack/repos/builtin/packages/cbtf-krell/package.py

```diff
@@ -66,12 +65,7 @@ class CbtfKrell(CMakePackage):
     depends_on("binutils@2.32")
 
     # For boost
-    depends_on("boost@1.70.0:")
-
-    # TODO: replace this with an explicit list of components of Boost,
-    # for instance depends_on('boost +filesystem')
-    # See https://github.com/spack/spack/pull/22303 for reference
-    depends_on(Boost.with_default_variants)
+    depends_on("boost@1.70.0:+filesystem+graph+program_options+python+test+thread")
 
     # For Dyninst
     depends_on("dyninst@10.1.0", when="@develop")
```

Comment:
@hainest: As you are working on `dyninst` and this depends on `dyninst`:

It looks like when using gcc-12.3.0 and `dyninst` uses `intel-tbb-2020.3`, maybe dyninst (intel-tbb) needs some fixing or conflict? I saw that some other package also has gotten a fix for `tbb:task` vs `task` (for gcc-13):

https://github.com/bambulab/BambuStudio/pull/1882/files#diff-40df5e0a2bcbc0ed23e2af22801fff02e39b9cb7bdefdd7af415d909d00024b0R7-R8
```py
                 from .stage/spack-stage-dyninst-10.1.0-dvcmvyczl7iuoddxin5sqfq4l5zfdq2l/spack-src/symtabAPI/src/dwarfWalker.C:48:
opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/intel-tbb-2020.3-7pmy6kavg6zc343p6owvrwjwzzmbn2m2/include/tbb/task.h:300:20: error: declaration of 'tbb:task& tbb:internal:task_prefix:task()' changes meaning of 'task' [-Wchanges-meaning]
  300 |         tbb:task& task() {return *reinterpret_cast<tbb:task*>(this+1);}
      |                    ^~~~
make: *** [Makefile:139: all] Error 2
==> Error: ProcessError: Command exited with status 2:
    opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gmake-4.4.1-krhvoy5tc5wo3p2bpim66lm35gbmykuj/bin/make -j64

             PI/src/dwarfWalker.C:48:
  >> 952     opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/intel-tbb-2020.3-7pmy6kavg6zc343p6owvrwjwzzmbn2m2/include/t
             bb/task.h:300:20: error: declaration of 'tbb:task& tbb:internal:task_prefix:task()' changes meaning of 'task' [-Wchanges-mea
             ning]
     1072          |             ^~~~
  >> 1073    make[2]: *** [symtabAPI/CMakeFiles/symtabAPI.dir/build.make:443: symtabAPI/CMakeFiles/symtabAPI.dir/src/dwarfWalker.C.o] Error 1
     1074    make[2]: *** Waiting for unfinished jobs....
     1075    .stage/spack-stage-dyninst-10.1.0-dvcmvyczl7iuoddxin5sqfq4l5zfdq2l/spack-src/symtabAPI/src/parseStab.C: In
             -dvcmvyc'
  >> 1105    make[1]: *** [CMakeFiles/Makefile2:762: symtabAPI/CMakeFiles/symtabAPI.dir/all] Error 2
     1106    make[1]: *** Waiting for unfinished jobs....
     1107    .stage/spack-stage-dyninst-10.1.0-dvcmvyczl7iuoddxin5sqfq4l5zfdq2l/spack-src/instructionAPI/src/Instructio
             -dvcmvyc'
  >> 1128    make: *** [Makefile:139: all] Error 2
```

Reply:
Sorry, I missed this note. I'll do some local testing to see what's going on.

Reply:
I wasn't able to replicate using the below and applying #47339. That patch is for a missing std include, so I wouldn't think it would affect TBB.

```
docker run --rm -it ubuntu:24.04
apt update -y
apt install -y gcc-12 g++-12 git cmake xz-utils bzip2 python3
git clone --depth=1 https://github.com/spack/spack
alias spack='/spack/bin/spack'
spack compiler find
spack external find --not-buildable python cmake
spack bootstrap now
spack install dyninst@10.1.0 %gcc@12.3.0 ^intel-tbb@2020.3
```